src/shared/tpm2-util.c

```diff
@@ -7489,17 +7641,25 @@ int tpm2_parse_luks2_json(
         if (!w)
                 return log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "TPM2 token data lacks 'tpm2-blob' field.");
 
-        r = json_variant_unbase64_iovec(w, &blob);
+        struct iovec *blobs = NULL;
+        size_t n_blobs = 0;
+        CLEANUP_ARRAY(blobs, n_blobs, iovec_array_free);
+
+        r = tpm2_parse_shard_array(w, "tpm2-blob", json_variant_unbase64_iovec, &blobs, &n_blobs);
         if (r < 0)
-                return log_debug_errno(r, "Invalid base64 data in 'tpm2-blob' field.");
+                return r;
```

Comment:
Maybe, better to keep and update logging? tpm2_parse_shard_array() does not log anything.

Reply:
Sorry, confused with tpm2_make_shard_array().